src/builder/create_attachment.rs

```diff
@@ -23,50 +29,45 @@ use crate::model::id::AttachmentId;
 pub struct CreateAttachment<'a> {
     pub filename: Cow<'static, str>,
     pub description: Option<Cow<'a, str>>,
-    pub data: Bytes,
+    pub data: AttachmentData<'a>,
 }
 
 impl<'a> CreateAttachment<'a> {
     /// Builds an [`CreateAttachment`] from the raw attachment data.
     pub fn bytes(data: impl Into<Bytes>, filename: impl Into<Cow<'static, str>>) -> Self {
         CreateAttachment {
-            data: data.into(),
             filename: filename.into(),
             description: None,
+            data: AttachmentData::Bytes(data.into()),
         }
     }
 
     /// Builds an [`CreateAttachment`] by reading a local file.
     ///
     /// # Errors
     ///
-    /// [`Error::Io`] if reading the file fails.
-    pub async fn path(path: impl AsRef<Path>) -> Result<Self> {
-        async fn inner(path: &Path) -> Result<CreateAttachment<'static>> {
-            let mut file = File::open(path).await?;
-            let mut data = Vec::new();
-            file.read_to_end(&mut data).await?;
-
-            let filename = path
-                .file_name()
-                .ok_or_else(|| std::io::Error::other("attachment path must not be a directory"))?;
-
-            Ok(CreateAttachment::bytes(data, filename.to_string_lossy().into_owned()))
-        }
-
-        inner(path.as_ref()).await
+    /// Returns [`Error::Io`] if the path is not a valid file path.
+    pub fn path(path: impl AsRef<Path>) -> Result<Self> {
+        let path = path.as_ref().to_owned();
```

Comment:
This is not good, just take `PathBuf` if you need it or store the `&Path` in `Self`.

Reply:
For some reason I had run into borrowck problems before, but no longer.

Reply:
This does mean that passing string literals will require the caller explicitly do a `.as_ref()`.

Reply:
I'm good with taking `impl AsRef<Path>` instead of `&Path`, the main concern was taking a reference to Path when a PathBuf is needed.

Reply:
Unfortunately that would cause lifetime problems since I'd be storing a reference to a `Path` that lives inside the current stack frame.

Reply:
Oh that's super annoying, but I don't see a nice workaround, good with merging like this.